From c8031fb7457f2e044e4f8b088d6974b68701521a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Iv=C3=A1n=20Budnik?= <ivanbuper@google.com>
Date: Mon, 20 Jun 2022 10:36:28 +0000
Subject: [PATCH] Fix Out of Bounds read in TextDescriptions.cpp

Fixing vulnerability in extract3GGPGlobalDescriptions() in
TextDescriptions.cpp

Bug: 233735886
Test: Run related PoC. See bug.
Change-Id: I87955b911d0a40390755321d332a11ecc9b20354
(cherry picked from commit b63d4e785ba4d896bbbd50d4f09bda13294926af)
Merged-In: I87955b911d0a40390755321d332a11ecc9b20354
---
 media/libstagefright/timedtext/TextDescriptions.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media/libstagefright/timedtext/TextDescriptions.cpp b/media/libstagefright/timedtext/TextDescriptions.cpp
index 2c2d11d45..3fec9edf6 100644
--- a/media/libstagefright/timedtext/TextDescriptions.cpp
+++ b/media/libstagefright/timedtext/TextDescriptions.cpp
@@ -466,6 +466,10 @@ status_t TextDescriptions::extract3GPPGlobalDescriptions(
 
                 if (subChunkType == FOURCC('f', 't', 'a', 'b'))
                 {
+                    if(subChunkSize < 8) {
+                        return OK;
+                    }
+
                     tmpData += 8;
                     size_t subChunkRemaining = subChunkSize - 8;
 
-- 
2.37.1.359.gd136c6c3e2-goog

